stuff/pipelines/client_contest.py

```diff
@@ -0,0 +1,146 @@
+    queue_name = result.method.queue
+    assert queue_name is not None
+    channel.queue_bind(
+        exchange='certificates',
+        queue=queue_name,
+        routing_key="ContestQueue")
+    logging.info('[*] waiting for the messages')
+
+    def certificate_contests_callback(
+            _channel: pika.adapters.blocking_connection.BlockingChannel,
+            _method: pika.spec.Basic.Deliver,
+            _properties: pika.spec.BasicProperties,
+            body: bytes) -> None:
+        data = json.loads(body.decode())
+        client = omegaup.api.Client(api_token=args.api_token, url=args.url)
+        scoreboard = client.contest.scoreboard(contest_alias=data['alias'],
+                                               token=data['scoreboard_url'])
+        ranking = scoreboard['ranking']
+        certificates: List[Tuple[str, int, str, Optional[int], str]] = []
+
+        for user in ranking:
+            contest_place: Optional[int] = None
+            if (data['certificate_cutoff']
+                    and user['place'] <= data['certificate_cutoff']):
+                contest_place = user['place']
+            verification_code = generate_code()
+            certificates.append((
+                'contest',
+                int(data['contest_id']),
+                verification_code,
+                contest_place,
+                str(user['username'])
+            ))
+        while True:
+            try:
+                cur.executemany('''
+                    INSERT INTO
+                        `Certificates` (
+                            `identity_id`,
+                            `certificate_type`,
+                            `contest_id`,
+                            `verification_code`,
+                            `contest_place`)
+                    SELECT
+                        `identity_id`,
+                        %s,
+                        %s,
+                        %s,
+                        %s
+                    FROM
+                        `Identities`
+                    WHERE
+                        `username` = %s;
+                    ''', certificates)
+                dbconn.commit()
+                break
+            except:  # noqa: bare-except
+                certificates = []
+                for user in ranking:
+                    contest_place = None
+                    if (data['certificate_cutoff']
+                            and user['place'] <= data['certificate_cutoff']):
+                        contest_place = user['place']
+                    verification_code = generate_code()
+                    certificates.append((
+                        'contest',
+                        int(data['contest_id']),
+                        verification_code,
+                        contest_place,
+                        str(user['username'])
+                    ))
+                logging.exception(
+                    'At least one of the verification codes was conflict')
```

Comment:
```suggestion
                    'At least one of the verification codes had a conflict')
```

aquí hay que decir que tuvo, en vez de hubo, conflicto.

Reply:
Listo